server/src/main/java/io/crate/replication/logical/MetadataTracker.java

```diff
@@ -284,6 +294,129 @@ static ClusterState updateIndexMetadata(String subscriptionName,
         }
     }
 
+    /**
+     * Extract new relations of all publications of the subscription based on the local and publishers cluster state
+     * and initiate a restore of these.
+     * See also {@link LogicalReplicationService#restore}.
+     */
+    private static void subscribeToNewRelations(String subscriptionName,
+                                                ClusterState subscriberClusterState,
+                                                ClusterState publisherClusterState,
+                                                LogicalReplicationService replicationService) {
+        var subscription = replicationService.subscriptions().get(subscriptionName);
+        assert subscription != null : "Cannot retrieve subscription '" + subscriptionName + "' from local cluster state";
+        subscribeToNewRelations(
+            subscriptionName,
+            subscriberClusterState,
+            publisherClusterState,
+            relationNames -> replicationService.updateSubscriptionState(
+                subscriptionName,
+                relationNames,
+                Subscription.State.INITIALIZING,
+                null
+            ),
+            (relationNames, toRestoreIndices, toRestoreTemplates) -> replicationService.restore(
+                subscriptionName,
+                subscription.settings(),
+                relationNames,
+                toRestoreIndices,
+                toRestoreTemplates
+            )
+        );
+    }
+
+    @VisibleForTesting
+    static CompletableFuture<Boolean> subscribeToNewRelations(
+        String subscriptionName,
+        ClusterState subscriberClusterState,
+        ClusterState publisherClusterState,
+        Function<Collection<RelationName>, CompletableFuture<Boolean>> updateStateFunction,
+        TriConsumer<Collection<RelationName>, List<String>, List<String>> restoreConsumer) {
```

Comment:
Minor: I prefer one method argument per line if formatted liked this.

```suggestion
        static CompletableFuture<Boolean> subscribeToNewRelations(
            String subscriptionName,
            ClusterState subscriberClusterState,
            ClusterState publisherClusterState,
            Function<Collection<RelationName>, 
            CompletableFuture<Boolean>> updateStateFunction,
            TriConsumer<Collection<RelationName>, 
            List<String>, 
            List<String>> restoreConsumer) {
```

Reply:
it is one argument per line, `restoreConsumer` is one argument with a unfortunately long definition ;)

Reply:
ups my bad.